### DISCUSSION TITLE
Can't proceed to Setup with Docker (Simple) - 500 - Server Error.

### DISCUSSION THREAD

**Discussion by Tiritibambix** [Kovah/LinkAce#593] [Support] [answered]
Hello. I was willing to install Linkace following the [simple setup](https://www.linkace.org/docs/v1/setup/setup-with-docker/simple/).

I follow these steps:
- edit docker-compose.yml which now looks like [this](https://pastebin.com/embed_js/Jg5nCsmR)
- edit .env which now looks like [this](https://pastebin.com/QdGcSWbT)
- ```chmod 666 .env```
- docker-compose up -d
- wait for ```mysqld: ready for connections```
- ```docker exec -it linkace_app_1 php artisan key:generate``` to generate the app key
- restart containers

All I get is ```500 - Server Error.
An internal server error occured. If you are the administrator, consult the application logs for details.```

WTH am I doing wrong ?

edit: logs are [here](https://pastebin.com/eLpTJeCn)(app) and [here](https://pastebin.com/2PU6s1Q3)(db)

**Comment by Kovah** [comment]
Could you take a look at the logs inside the LinkAce container? They are stored under /app/storage/logs.

**Reply by Tiritibambix** [reply]
The folder is empty. That's why I had to ```docker container logs -f linkace_app_1``` and ```docker container logs -f linkace_db_1``` to get the 2 logs linked in my first post.

**Comment by Tiritibambix** [comment, marked as answer]
I tried a different approach. I cloned the entire repo and used ```.env.docker``` and ```docker-compose.production-simple.yml```.

I changed as few things as possible : the passwords in ```.env.docker```, and volumes path (+commented backup volume path) in ```docker-compose.production-simple.yml```

Logs folder is still empty, but I end up with a red page and a lot of errors.
And I got [**this**](https://clip.steph.click/!PbLtrnqXyP) in "Stack Trace"

![LAMerge](https://user-images.githubusercontent.com/69852851/211040673-56a81900-621d-4d84-b91f-dffd5dcac2aa.jpg)

**Reply by Kovah** [reply]
If you clone the repo, make sure that the `storage/logs` directory is writable by your web server / php process.

**Reply by Tiritibambix** [reply]
So I tried installing without binding ```/var/lib/mysql``` or ```/app/storage/logs```to any specific location on my drive and left it unmodified like this:

```
    volumes:
      - db:/var/lib/mysql
[..]
    volumes:
      - linkace_logs:/app/storage/logs
```
I'm pretty new when it comes to this. How can I give permission so that I can bind these volumes to a specific place like this :

```
    volumes:
      - /location/on/my/drive/for/linkace/db:/var/lib/mysql
[..]
    volumes:
      - /location/on/my/drive/for/linkace/linkace_logs:/app/storage/logs
```

**Reply by Kovah** [reply]
In that case `/location/on/my/drive/for/linkace/linkace_logs` should probably be writable.

**Reply by Tiritibambix** [reply]
That's nuts. Binding to some specific location on my drive works great with my multiple other applications. It only is an issue with LinkAce.

Would adding PUID / PGID in the variables help somehow ?

**Reply by Kovah** [reply]
If there is no specific reason to keep the logs outside of Docker, I would rather recommend using a volume. Permissions are always a problem and using other users for the containers will lead to other issues.

**Reply by Tiritibambix** [reply]
> If there is no specific reason

But there is :D That would be data persistence and backup convenience.

I am still unsure. Am I doing things wrong, or is it a limitation imposed by LinkAce itself ?

**Reply by Kovah** [reply]
The simplest solution that I use too is to grant write permissions on that folder on the host where the data needs to be stored.
The limitation is not directly LinkAce itself but the whole Docker thing. I just don't want to spend time preparing the Docker image so that arbitrary user IDs can be used with it when there are simple solutions.

**Reply by Tiritibambix** [reply]
> grant write permissions on that folder on the host where the data needs to be stored.

I hate to ask (but I guess this will be my last question), but what's the process to accomplish that ?

Edit: after searching, I just ```sudo chmod o+w _folder_``` and it worked.
If there is a more elegant way to do this, please let me know.

**Reply by Kovah** [reply]
chmod +w _folder_ is correct. hope that LinkAce works now. 😬

**Reply by Tiritibambix** [reply]
It does !!!

Thanks a lot for your work and for your time :)
